### ISSUE TITLE
The component Styled(FaTimes) with the id of "sc-cxFLnm" has been created dynamically. You may see this warning because you've called styled inside another component. To resolve this only create new StyledComponents outside of any render method and function component. 

### ISSUE THREAD

**Issue by capndave** [styled-components/styled-components#3539]
I could use some help. None of the threads on stack overflow use the `styled(WrappedComponent)` syntax. How do I use the below child component without throwing an error? There is a [similar issue](https://github.com/styled-components/styled-components/issues/3117) to this which is either not well documented or unresolved

## Child Component
```javascript
import styled from "styled-components"
import { FaAngleDown } from "react-icons/fa"

export const StyledDownArrow = styled(FaAngleDown)`
  color: black;
  font-size: 120%;
  margin-left: -2rem;
  margin-bottom: -0.25rem;
`
```

## Parent Component
```javascript
import React, { useState } from "react"
import { DownArrow } from "../../Arrows

export const ParentComponent = () => <div>look at me! <DownArrow /></div>
```

## Error Message
```bash
The component Styled(FaTimes) with the id of "sc-cxFLnm" has been created dynamically. You may see this warning because you've called styled inside another component. To resolve this only create new StyledComponents outside of any render method and function component.
```

## Additionally
I noticed that the error message lists the svg component as FaTimes rather than FaAngleDown. Could that have some effect on this?

**Comment by jnthnk** [comment]
Hi. The icon imported from the package "react-icons/fa" has nothing to do with styled component. Is just a react component.

You can only use styled(Component) method only with styled-components.

A simple solution, can be simply wrap the FaAngleDown component inside a StyledIconWrapper styled-component and style the last one to apply to che child using css (& > svg)

**Comment by kitten** [comment]
with out a reproduction I cannot really help you here unfortunately. However, you can see how the warning itself is generated: https://github.com/styled-components/styled-components/blob/001957c2dfae92a289ab7b47b82875fba05e281f/packages/styled-components/src/utils/checkDynamicCreation.ts

You can tell here that it won't stop you wrong continuing, but it'll essentially output a warning telling you that you may have called `styled(X)` inside another component. This is based on calling a hook function from React, which we expect to fail if it's called outside of a component (which should always be the case).

So, while I can't investigate the exact cause of your warning, this may put you on a good path to solving it. If needed though, we can reopen and revisit this issue, especially if you think it's erroneous and have a reproduction :v:
